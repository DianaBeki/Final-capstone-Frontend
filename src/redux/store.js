import { configureStore } from '@reduxjs/toolkit';
import doctorReducer from './doctor/doctorSlice';
import userSlice from './user/userSlice';

const store = configureStore({
  reducer: {
    doctor: doctorReducer,
    user: userSlice,
  },
});

export default store;
