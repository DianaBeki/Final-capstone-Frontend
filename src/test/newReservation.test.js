import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import NewReservation from '../components/pages/newReservation';
import store from '../redux/store';

const render = (component) => rtlRender(
  <Provider store={store}>
    <Router>
      {component}
    </Router>
  </Provider>,
);

describe('NewReservation Component', () => {
  it('renders NewReservation component', () => {
    const { container } = render(<NewReservation />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
