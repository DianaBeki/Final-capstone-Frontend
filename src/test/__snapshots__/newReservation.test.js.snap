// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewReservation Component renders NewReservation component 1`] = `
<div
  class="flex flex-col items-center justify-center min-h-screen custom-styles"
>
  <div
    class="container mx-auto max-w-screen-md p-8 reserve-div1"
  >
    <h2
      class="text-2xl font-bold mt-8 mb-4"
    >
      Reservation Schedule Information
    </h2>
    <hr
      class="reserve-hr"
    />
    <p
      class="text-gray-700"
    >
      Our doctors are available from Monday to Friday, 9:00 AM to 5:00 PM. If you have any questions or need assistance, feel free to contact our support team at support@example.com.
    </p>
  </div>
  <div
    class="container mx-auto max-w-screen-md p-8 reserve-div2"
  >
    <form
      class="flex flex-col space-y-4 md:flex-row md:space-y-0 md:space-x-4 w-full"
    >
      <div
        class="flex flex-col w-full md:w-1/2"
      >
        <p
          class="text-lg font-medium text-gray-600"
          style="color: white;"
        >
          Date
        </p>
        <input
          class="p-2 border rounded reserve-input"
          id="date"
          required=""
          type="date"
          value=""
        />
      </div>
      <div
        class="flex flex-col w-full md:w-1/2"
      >
        <p
          class="text-lg font-medium text-gray-600"
          style="color: white;"
        >
          City
        </p>
        <input
          class="p-2 border rounded reserve-input"
          id="city"
          required=""
          type="text"
          value=""
        />
      </div>
      <div
        class="flex flex-col w-full md:w-1/2"
      >
        <p
          class="text-md font-medium text-gray-600"
          style="color: white;"
        >
          Choose Doctor
        </p>
        <select
          class="p-2 border rounded reserve-select"
          id="doctor"
          required=""
        >
          <option
            value=""
          >
            Select Doctor
          </option>
        </select>
      </div>
      <button
        class="reserve-btn "
        type="submit"
      >
        Reserve
      </button>
    </form>
  </div>
</div>
`;
